pkg/sink/sink.go

```diff
@@ -119,6 +129,10 @@ func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
 			code = thiscode
 			break
 		}
+		if thiscode == http.StatusBadRequest && code != http.StatusCreated {
+			code = http.StatusBadRequest
+			errMsg = marshalErr
```

Comment:
This doesn't layer well with the unauthorized error- if the next trigger is unauthorized, you'll have an unauthorized error with a marshal error detail.

Reply:
I put errMsg = "" if unauthorized error comes. So we won't show that.